src/Component/Console/src/Helper/Table.js

```diff
@@ -0,0 +1,633 @@
+const InvalidArgumentException = Jymfony.Component.Console.Exception.InvalidArgumentException;
+const Helper = Jymfony.Component.Console.Helper.Helper;
+const OutputInterface = Jymfony.Component.Console.Output.OutputInterface;
+
+class Table {
+    /**
+     * @param {OutputInterface} output An OutputInterface instance
+     * @private
+     */
+    __construct(output) {
+        /**
+         * Table headers.
+         *
+         * @type {String[]}
+         * @private
+         */
+        this._headers = [];
+
+        /**
+         * Table rows.
+         *
+         * @type {Array}
+         * @private
+         */
+        this._rows = [];
+
+        /**
+         * Number of columns cache.
+         *
+         * @type {int}
+         * @private
+         */
+        this._numberOfColumns = 0;
+
+        /**
+         * Column widths cache.
+         *
+         * @type {Object}
+         * @private
+         */
+        this._effectiveColumnWidths = [];
+
+        /**
+         * @type {Array}
+         * @private
+         */
+        this._columnStyles = [];
+
+        /**
+         * User set column widths.
+         *
+         * @var {Array}
+         * @private
+         */
+        this._columnWidths = [];
+
+        /**
+         * @type {OutputInterface}
+         * @private
+         */
+        this._output = output;
+
+        if (!this.styles) {
+            Table.styles = Table.initStyles();
+        }
+
+        this.style = 'default';
+    }
+
+    static get styles() {
+        return Table.styles;
```

Comment:
Infinite recursion. Use local variable to statically store styles.

Reply:
:+1: